hls4ml/converters/onnx_to_hls.py

```diff
@@ -115,7 +116,64 @@ def compute_pads_2d(operation, layer):
     
     return pads
 
-def onnx_to_hls(yamlConfig):
+####----------------------Optimizer---------------------######
+def _hls4ml_onnx_optimizer(graph):
```

Comment:
Why is this not implemented as a regular hls4ml optimizer? `Transpose` is a valid layer and should appear in the internal model, then if it is placed before and after a real layer (which only happens when using one very poorly designed ONNX conversion tool) we can optimize it away.

Reply:
Sorry, I'm not sure what you mean by this? so do suggest that I implement it [here](https://github.com/fastmachinelearning/hls4ml/tree/master/hls4ml/model/optimizer) instead of an onnx optimizer?

Reply:
What ONNX models have this issue? PyTorch or Keras? If the issue is with PyTorch, does [this](https://pytorch.org/tutorials/intermediate/memory_format_tutorial.html) help?

Performing graph transformations in the converter will make it harder to maintain it in the future. We've been there before. For example, your current transformation is dangerous, you need to make extra checks to ensure that the axes of transpose match (meaning you are transposing back into original format). Also how would you add future optimizations on top of this? You're modifying the graph as you iterate it, that's asking for trouble.